lib_test/bap_types/test_const_prop.ml

```diff
@@ -0,0 +1,154 @@
+open Core_kernel.Std
```

Comment:
Your test only verifies that the propagation, i.e., substitution of a value occurs, it doesn't verify the most important property - that the semantics of the program is preserved. 

Your tests must also verify that your transformation is sound and complete, i.e., that for all programs `p`, `p'` such that `p' = f(p)` the following is true:

soundness: `P(p) -> P(p')`
completeness: `P(p') -> P(p)`

for all  predicates  `P` from a set of predicates that defines the semantics of BIL.